### PR TITLE
Pre enable

### PR DESCRIPTION
**Pull request by rbiasini**
Switching to "rising edge" cancellation type, to allow the enabling when: 
- gas pedal is pressed
- brake pedal is pressed and speed is zero

### CODE REVIEW

File: board/safety/safety_honda.h [hunk 1]
```diff
@@ -3,51 +3,80 @@
...
-  // exit controls on brake press
+  // exit controls on rising edge of brake press or on brake press when
+  // speed > 0
   if ((to_push->RIR>>21) == 0x17C) {
+
     // bit 53
-    if (to_push->RDHR & 0x200000) {
+    int brake = to_push->RDHR & 0x200000;
+
+    if (brake && !(brake_prev)) {
```
Comment (geohot) [inline board/safety/safety_honda.h]: I would refactor this to 

if (brake && (!(brake_prev) || speed)) {
  controls_allowed = 0
}

File: board/safety/safety_honda.h [hunk 2]
```diff
@@ -3,51 +3,80 @@
...
+  // sample speed
+  if ((to_push->RIR>>21) == 0x158) {
+    // first 2 bytes
+    speed = to_push->RDLR & 0xFFFF;
+  }
+
   // state machine to enter and exit controls
   // 0x1A6 for the ILX, 0x296 for the Civic Touring
   if ((to_push->RIR>>21) == 0x1A6 || (to_push->RIR>>21) == 0x296) {
+
```
Comment (geohot) [inline board/safety/safety_honda.h]: Why were these lines added?

Comment (rbiasini) [reply board/safety/safety_honda.h]: fixed

File: board/safety/safety_honda.h [hunk 3]
```diff
@@ -25,37 +33,57 @@ static void honda_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
...
       }
+      gas_prev = gas;
     }
   }
+
+  // block actuator commands if gas or brake (with vehicle moving) are pressed
+  if (gas_prev || gas_interceptor_prev || (brake_prev && speed)) {
+    actuators_blocked = 1;
+  } else {
+    actuators_blocked = 0;
```
Comment (geohot) [inline board/safety/safety_honda.h]: actuators_blocked = !controls_allowed;

File: board/safety.h [hunk 4]
```diff
@@ -16,6 +16,7 @@ typedef struct {
 
 // This can be set by the safety hooks.
 int controls_allowed = 0;
+int actuators_blocked = 1;
```
Comment (geohot) [inline board/safety.h]: Move to safety_honda
